Add font tarballs to repository

xjuan opened the pull request:
This avoids checksum errors everytime a font changes upstream.
Use fonts/download_fonts.sh to update fonts when desired.

https://phabricator.endlessm.com/T19905

ebassi commented:
It would be neat if you could modify the script that generates the manifest to automatically get the SHA checksum of each font package and fill up the `sha256` field in the JSON file.

ebassi commented on generate-manifest.py:
```diff
@@ -76,6 +77,30 @@ def edit_manifest(data, arch, branch, runtime_version):
+                             'dest-filename': font,
+                             'sha256': sha256(path)})
+
+    data['modules'].append({ 'name': 'default-theme-fonts',
```
I'd rather have `data['modules'].prepend()` here, for two reasons:

 - early failure
 - keep the `os-release` module at the end, as it's the last part of the build process

ebassi commented on fonts/download_fonts.sh:
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -e
+
+GOOGLE_FONTS=(Fira+Sans Lato Libre+Baskerville Merriweather Noto+Sans Noto+Serif Raleway Roboto)
```
Can we add this as a `Makefile` rule, instead?

ebassi commented on generate-manifest.py:
```diff
@@ -76,6 +77,30 @@ def edit_manifest(data, arch, branch, runtime_version):
+def add_fonts_module(data):
+    """Add fonts module to manifest json"""
+
+    sources = [ { 'type': 'file', 'path': 'fonts-makefile', 'dest-filename': 'Makefile' } ]
```
Since we're updating the manifest, I think we should also drop the `fonts-makefile` hack, and use `"build": "simple"`.

Commit:
Add font tarballs to repository

This avoids checksum errors everytime a font changes upstream.
Use new makefile rule update-fonts to update fonts from google fonts servers.

https://phabricator.endlessm.com/T19905

ptomato commented on generate-manifest.py:
```diff
@@ -76,6 +77,34 @@ def edit_manifest(data, arch, branch, runtime_version):
 
+def sha256(filename):
+    checksum = hashlib.sha256()
+    with open(filename, 'rb') as file:
```
For next time: `file` is a built-in in Python, so it's better not to override it

ptomato reviewed (approved):
Looks good, one style nitpick but no need to block the pull request on it.
